js/controller/accountcontroller.js

```diff
@@ -33,23 +33,37 @@ define(function(require) {
 		var defer = $.Deferred();
 		var fetchingAccounts = Radio.account.request('entities');
 
+		// Do not show sidebar content until everything has been loaded
+		Radio.ui.trigger('sidebar:loading');
+
 		$.when(fetchingAccounts).done(function(accounts) {
 			if (accounts.length === 0) {
 				defer.resolve(accounts);
 				Radio.navigation.trigger('setup');
+
+				Radio.ui.trigger('sidebar:accounts');
 			} else {
 				var loadingAccounts = accounts.map(function(account) {
+					require('state').accounts.add(account);
```

Comment:
- [ ] TODO: the account might already be in the collection, hence we can remove that method call